### ISSUE TITLE
Help/Discussion: Attempting to automate screenshots using f3d - cannot get camera position to desired view

### ISSUE THREAD

**Issue by nathaneltitane** [f3d-app/f3d#542]
Hello,

This is not a bug but rather a query for help:

I am attempting to automate via BASH scripting the rendering and grabbing of stl model files and cannot, for the life of me get the camera to show the model appropriately. The camera always seems to focus or present 'past' the model (i.e. behind it) resulting in a blank screen, until i scroll back to show the loaded model.

here is the current CLI command i am running:

f3d --camera-focal-point=2,2,2 --camera-position=2,2,2 --camera-elevation-angle=0 --camera-view-angle=45 --resolution 2048,2048 -xqat ${MODEL_FILE}.stl

and here is a sample image of what i am trying to achieve:

![image](https://user-images.githubusercontent.com/5674757/208315907-7d9c6112-8912-4a2e-b1ab-da8806e94240.png)

n.b.: the camera position seems to be ok, but the focus seems to be set waaaay past the model boundary.

Thank you

**Comment by Meakk** [comment]
```
--camera-focal-point=2,2,2 --camera-position=2,2,2
```

This is incorrect, the camera position, and the focal point cannot be the same. The focal point is used to determine the direction of the camera.
Usually the focal point is the center of the bounding volume of your model, or simply the origin (0,0,0).
You probably need to declare the camera up vector. In your case it's +Y, so (0,1,0).

Not that you can press `?` in F3D to dump the current camera position to give you an idea of the values you have to set.

**Comment by nathaneltitane** [comment]
> ```
> --camera-focal-point=2,2,2 --camera-position=2,2,2
> ```
> 
> This is incorrect, the camera position, and the focal point cannot be the same. The focal point is used to determine the direction of the camera. Usually the focal point is the center of the bounding volume of your model, or simply the origin (0,0,0). You probably need to declare the camera up vector. In your case it's +Y, so (0,1,0).
> 
> Note that you can press `?` in F3D to dump the current camera position to give you an idea of the values you have to set.

thank you for your prompt reply - seems that dumping the info helped, but now i have a weird cutoff with the newly set parameters? could it be the camera fov?

f3d --camera-view-up=0,1,0 --camera-position=240,240,240  --camera-elevation-angle=0 --camera-view-angle=45 --resolution 2048,2048 -xqat /media/internal/Printer/models/pew-pew-prints/battery-bay-adventure-force-spectrum.stl &

![image](https://user-images.githubusercontent.com/5674757/208318689-d496c56d-5927-4f39-8a27-c48666413f04.png)

**Comment by Meakk** [comment]
The cutoff is most likely a bug in the clipping range update.  
Can you share a STL file with the exact command you are running in order to reproduce the issue?

Regarding the camera position relative to the bounding box, it's not supported yet. Unfortunately, you have to know in advance the bounding volume to compute a position that makes sense.

**Comment by nathaneltitane** [comment]
> The cutoff is most likely a bug in the clipping range update. Can you share a STL file with the exact command you are running in order to reproduce the issue?
> 
> Regarding the camera position relative to the bounding box, it's not supported yet. Unfortunately, you have to know in advance the bounding volume to compute a position that makes sense.

when scrolling out and back to position (using ? as output reference) the cutoff disappears (far field).

judging by your answer, if i base my render settings, for the time being, off of the biggest model i have, i guess things should work?

can I mail you the stl file @Meakk ?

**Comment by Meakk** [comment]
Sure. Can you join the discord? I'll give you my email in private.

**Comment by nathaneltitane** [comment]
> Sure. Can you join the discord? I'll give you my email in private.

Trying, but i'M getting a stupid verification loop - my email is on my GH, send me a message?

**Comment by Meakk** [comment]
@nathaneltitane here is my result:

Command:
```
f3d --camera-position=240,240,240 --resolution 2048,2048 -xqat --dry-run ~/Downloads/magazine-well-wall-hanger-long-dart-horizontal-command-strip.stl --output=/tmp/out.png
```

Output file:
![out](https://user-images.githubusercontent.com/6069752/208780108-d3daa1eb-d1e9-4bb8-8e51-80067d3715b7.png)

Version:
```
$ f3d --version
f3d 1.3.1

F3D - A fast and minimalist 3D viewer
Version: 1.3.1.
Build date: 2022-12-16 20:47:45.
Build system: Linux x86_64.
Compiler: Clang 14.0.6.
External rendering module: ON.
Raytracing module: ON.
Exodus module: ON.
OpenCASCADE module: 7.6.3 (full support).
Assimp module: 5.2.4.
Alembic module: 1.8.4.
VTK version: 9.1.0 (build 0).
Copyright (C) 2019-2021 Kitware SAS.
Copyright (C) 2021-2022 Michael Migliore, Mathieu Westphal.
License BSD-3-Clause.
By Michael Migliore, Mathieu Westphal and Joachim Pouderoux.
```

**Comment by nathaneltitane** [comment]
used camera azimuth and elevation angles with no background and direct output to file. solved!

f3d --no-background --output=$HOME/${MODEL_RENDER}.png --camera-azimuth-angle=30 --camera-elevation-angle=30  --resolution 2048,2048 -qat ${MODEL_FILE}.stl

![test](https://user-images.githubusercontent.com/5674757/208786713-87df679c-233b-42ac-8072-5cf67d8e1ebf.png)

**Comment by nathaneltitane** [comment]
Hello!

Seem to have a similar issue after using the mentioned parameters:

Tall and skinny part generates a clipping and when enabling the grid there is no fade not colored origin lines...

![image](https://user-images.githubusercontent.com/5674757/217714393-3e0b504c-328a-4019-975d-20f5dc121044.png)

**Comment by mwestphal** [comment]
Could you share the file ?

**Comment by nathaneltitane** [comment]
> 

discord?

**Comment by mwestphal** [comment]
Looking ok here, which command are you using ?


![a](https://user-images.githubusercontent.com/3129530/217727442-9335bd0e-a984-4a1a-8d97-a21a99271a68.png)

**Comment by nathaneltitane** [comment]
> Looking ok here, which command are you using ?
> 
> ![a](https://user-images.githubusercontent.com/3129530/217727442-9335bd0e-a984-4a1a-8d97-a21a99271a68.png)

i use my automation script to produce output/renders to a directory with a specific set of parameters, but also as hinted by our previous chats.

--
pew_pew_render () {
	# render model

	frobulator.inf "Rendering" "[ ${view[coordinates]} // ${model_name} ]"
	echo

	frobulator.fwd "Output" "[ ${render_file} ]"
	echo

	model_path="${1}"
	coordinates="${2}"

	if [[  -z "${height}" ]]
	then
		height="4096"
	fi

	if [[ -z "${width}" ]]
	then
		width="4096"
	fi

	extension="png"

	f3d																	\
		--ssao															\
		--fxaa															\
		--tone-mapping													\
		--resolution "${width}","${height}"								\
		--camera-azimuth-angle=${latitude[$coordinates]}				\
		--camera-elevation-angle=${longitude[$coordinates]}				\
		--no-background													\
		--output="${model_name}-${view[$coordinates]}.${extension}"		\
																		"${model_path}" > /dev/null 2>&1

		wait
}

--

**Comment by mwestphal** [comment]
Fixed by switching to 1.3.1
